crates/networking/p2p/net.rs

```
@@ -331,7 +359,96 @@ async fn discover_peers_server(
                     }
                 }
             }
-            _ => {}
+            Message::ENRRequest(msg) => {
+                if is_expired(msg.expiration) {
+                    debug!("Ignoring enr-request msg as it is expired.");
+                }
+                // Note we are passing the current timestamp as the sequence number
+                // This is because we are not storing our local_node updates in the db
+                let Ok(node_record) = NodeRecord::from_node(local_node, time_now_unix(), &signer)
+                else {
+                    debug!("Ignoring enr-request msg could not build local node record.");
+                    continue;
+                };
+                let msg = discv4::Message::ENRResponse(ENRResponseMessage::new(
+                    packet.get_hash(),
+                    node_record,
+                ));
+                let mut buf = vec![];
+                msg.encode_with_header(&mut buf, &signer);
+                let _ = udp_socket.send_to(&buf, from).await;
```

Should we return an error instead?

---

Because this is inside the `discovery` loop returning an error would close the task. We don't care if we fail to send the message because there is nothing to do really. Thou I think is worth adding a debug message in case of an error.

---

> Thou I think is worth adding a debug message in case of an error.
Only if you find it worth it from the perspective of a person who will debug this if needed.

---

A more proper error handling is being addressed in #1768.